lib/preload.js

```diff
@@ -251,11 +265,19 @@ preload.CONTAINER_NAME = 'resin-image-preloader'
 
 /**
  * Build the preloader docker image
+ * @param {Object} docker - dockerode instance
+ * @param {Object} visuals - resin-cli-visuals (optional)
  * @returns {Stream}
  */
-preload.build = (docker) => {
+preload.build = (docker, visuals) => {
```

Comment:
For 1, personally I wouldn't worry. I agree it is slightly messy, but it doesn't actually cause any problems, and it's a pretty tiny module.

For 2, this is actually a big negative: if the versions diverge, that means this module is now going to get given an instance of visuals that has breaking changes in it, and may not behave in the way it expects. If we release a version of `resin-cli-visuals` that renames `visuals.Progress` to `visuals.ProgressBar`, this code will quietly and unexpectedly implode, because it doesn't say or check anywhere which version of visuals it's expecting. If we make this module depend on visuals directly instead, then we would end up with two copies, but that's a tiny problem, and everything keeps working in the meantime.

Does it cause problems if the standalone mode has nice visuals, or is it just that it's not useful? If it's not going to break anything, I think I'd just depend on it and enable visuals all the time: it solves the above problem, and nicely simplifies the code by letting you remove the `if (visuals)` everywhere too.